src/library/Uno.Material/Extensions/ResourceDictionaryExtensions.cs

```diff
@@ -0,0 +1,82 @@
+	{
+		/// <summary>
+		/// Used to safely nest a res-dict under another, regardless if it was nested before.
+		/// </summary>
+		/// <param name="rd"></param>
+		/// <param name="nested"></param>
+		/// <remarks>Used for <see cref="MaterialColors"> and <see cref="MaterialFonts"> "delay-one" initialization only.</remarks>
+		public static void SafeMerge(this ResourceDictionary rd, ResourceDictionary nested)
+		{
+			/* Here is an example of typically App.xaml setup:
+				<MaterialColors OverrideSource="ms-appx:///custom-colors.xaml" /> <!-- @1 -->
+				<MaterialFonts OverrideSource="ms-appx:///custom-fonts.xaml" />
+				<MaterialResources>
+					<!-- simplified for briefty, but the essence is the same if we expand it -->
+					<ResourceDictionary.MergedDictionaries>
+						<MaterialColors />  <!-- @2 -->
+						<MaterialFonts />
+					<ResourceDictionary.MergedDictionaries>
+
+					<!-- Resources/Styles... -->
+				</MaterialResources>
+			
+				<MaterialColors OverrideSource=... /> <!-- @3 -->
+				<MaterialToolkitResources /> <!-- same structure as MaterialResources that nested MaterialColors&Fonts -->
+			*/
+
+			// In order for the override to work, it has to be included within MaterialResource.MergedDictionary (see: @2),
+			// or otherwise the brush/font will be resolved with the default font/colors. Due to the way MaterialResource is setup,
+			// we don't direct access to the nested resources, early enough (is this necessary?).
+			// We workaround this by using a "delay-one" initialization hack. MaterialFonts/Colors contains a static field to hold the override dictionary.
+			// We can indirectly assign this field through OverrideSource dependency property. And in the .ctor, we merge this dictionary if available.
+			// And, the initialization sequence will look like the follow:
+			// 1. MaterialColors@1 is ctor'd, but the static field is still null.
+			// 2. MaterialColors@1 OverrideSource is set, and we store that to the static field.
+			// 3. MaterialColors@2 inside the MaterialResources is ctor'd, the static field is non-null, so we merge that.
+
+			// That works fine, except on uwp/winui-desktop it is illegal to nest the same instance of dictionary under multiple parent dictionary.
+			// That happens when we new up any instance of MaterialFonts/Colors after the "delay-one-init" hack has proc'd once directly or indirectly:
+			// 4. MaterialColors@3 is ctor'd, the static field IS NON-NULL, so we merge that AGAIN.
+
+			// While we can set the static field to null after merging it, it would break if we define MaterialToolkitResources after MaterialResources.
+			// The toolkit res would not receive the proper override, unless we define another MaterialColors with override.
+			// But that is not intuitive for the user and would require additional explantion.
+			// So to keep it simple, we will just always merged a clone instead of original instance (we have no way of determining if the dict is already child of another).
```

Comment:
```suggestion
			// So to keep it simple, we will just always merge a clone instead of original instance (we have no way of determining if the dict is already a child of another).
```